pkg/resources/execute.go

```diff
@@ -0,0 +1,172 @@
+	},
+	"revert": {
+		Type:        schema.TypeString,
+		Required:    true,
+		Description: "SQL statement to revert the execute statement. Invoked when resource is being destroyed.",
+	},
+	"query": {
+		Type:        schema.TypeString,
+		Optional:    true,
+		Description: "Optional SQL statement to do a read. Invoked on every resource refresh and every time it is changed.",
+	},
+	"query_results": {
+		Type:        schema.TypeList,
+		Computed:    true,
+		Description: "List of key-value maps (text to text) retrieved after executing read query. Will be empty if the query results in an error.",
+		Elem: &schema.Schema{
+			Type: schema.TypeMap,
+			Elem: &schema.Schema{
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+		},
+	},
+}
+
+func Execute() *schema.Resource {
+	return &schema.Resource{
+		CreateContext: TrackingCreateWrapper(resources.Execute, CreateExecute),
+		ReadContext:   TrackingReadWrapper(resources.Execute, ReadExecute),
+		UpdateContext: TrackingUpdateWrapper(resources.Execute, UpdateExecute),
+		DeleteContext: TrackingDeleteWrapper(resources.Execute, DeleteExecute),
+
+		Schema: executeSchema,
+		Importer: &schema.ResourceImporter{
+			StateContext: schema.ImportStatePassthroughContext,
+		},
+
+		Description: "Resource allowing execution of ANY SQL statement.",
+
+		CustomizeDiff: TrackingCustomDiffWrapper(resources.UnsafeExecute, customdiff.All(
+			customdiff.ForceNewIfChange("execute", func(ctx context.Context, oldValue, newValue, meta any) bool {
+				return oldValue != ""
+			}),
+			func(_ context.Context, diff *schema.ResourceDiff, _ any) error {
+				if diff.HasChange("query") {
+					err := diff.SetNewComputed("query_results")
+					if err != nil {
+						return err
+					}
+				}
+				return nil
+			}),
+		),
+	}
+}
+
+func CreateExecute(ctx context.Context, d *schema.ResourceData, meta any) diag.Diagnostics {
+	client := meta.(*provider.Context).Client
+
+	id, err := uuid.GenerateUUID()
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	executeStatement := d.Get("execute").(string)
+	_, err = client.ExecUnsafe(ctx, executeStatement)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	d.SetId(id)
+	log.Printf(`[INFO] SQL "%s" applied successfully\n`, executeStatement)
+
+	return ReadExecute(ctx, d, meta)
+}
+
+func UpdateExecute(ctx context.Context, d *schema.ResourceData, meta any) diag.Diagnostics {
+	if d.HasChange("query") {
+		return ReadExecute(ctx, d, meta)
+	}
+	return nil
+}
+
+func ReadExecute(ctx context.Context, d *schema.ResourceData, meta any) diag.Diagnostics {
+	client := meta.(*provider.Context).Client
+
+	readStatement := d.Get("query").(string)
+
+	setNilResults := func() diag.Diagnostics {
+		log.Printf(`[DEBUG] Clearing query_results`)
+		err := d.Set("query_results", nil)
+		if err != nil {
+			return diag.FromErr(err)
+		}
+		return nil
+	}
+
+	if readStatement == "" {
+		return setNilResults()
+	} else {
+		rows, err := client.QueryUnsafe(ctx, readStatement)
```

Comment:
Nit: this block does not need to be nested.

Reply:
let's leave it this way for now